test/backend.test.js

```diff
@@ -92,7 +94,22 @@ describe('Full walkthrough', function () {
   }
 
   function createSocket () {
-    return createWebSocket(process.env.API_URL, { timeout: 3000, strategy: false })
+    return new Promise((resolve, reject) => {
+      createGIPubSubClient(process.env.API_URL, {
+        handlers: {
+          error (event) {
+            reject(event)
+          },
+          open (event) {
+            resolve(this)
+          }
+        },
```

Comment:
The API is quite complex and confusing, so I'm not sure, but is this missing a `NOTIFICATION_TYPE.ENTRY` handler?

Reply:
I'm sorry if the API isn't simple enough. Is there something in particular that should better be removed?

Regarding these functions, they are `ws` socket event handlers; perhaps it was more obvious when using `onerror` and `onopen` as keys?

On the other hand, `NOTIFICATION_TYPE.ENTRY` denotes a pubsub message type.
Such a type constant can be used as a key in `messageHandlers` to define a processing routine ("message handler") that should be run whenever an `ENTRY` message is received.

It's still possible to not use the `handlers` options and just bind any event handler manually upon object instantiation.
Similarly, it's also possible to not use the `messageHandlers` option and just put any message processing logic in a rather large switch statement inside a `message` socket event handler.

Reply:
Ah, my apologies, please ignore my comment. I understand now better how this works and why `ENTRY` doesn't need to be specified here. It's because it is already specified as part of the implementation of `createGIPubSubClient`. That was just me getting confused and not looking closely enough. Resolving this.